### ISSUE TITLE
SKBitmap.Decode() not working with WEBP from remote stream

### ISSUE THREAD

**Issue by smartbracker** [mono/SkiaSharp#407]
I try download and  resize a remote image with format WEBP, but i found this issue

```ruby

            using (var httpClient = new HttpClient())
            using (var stream = await httpClient.GetStreamAsync(new Uri("http://www.gstatic.com/webp/gallery/2.webp")))
            using (var skBitmap = SKBitmap.Decode(stream))
            using (var destinationFile = new FileStream("D:/desktop/teste_webp_format.jpg", FileMode.OpenOrCreate))
            using (var resized = skBitmap.Resize(new SKImageInfo(100, 100), SKBitmapResizeMethod.Lanczos3))
            using (var image = SKImage.FromBitmap(resized))image.Encode(SKEncodedImageFormat.Jpeg, 100).SaveTo(destinationFile);
```

working fine with other image formats.

```ruby

            using (var httpClient = new HttpClient())
            using (var stream = await httpClient.GetStreamAsync(new Uri("https://www.gstatic.com/webp/gallery/4.sm.jpg")))
            using (var skBitmap = SKBitmap.Decode(stream))
            using (var destinationFile = new FileStream("D:/desktop/test_jpg_format.jpg", FileMode.OpenOrCreate))
            using (var resized = skBitmap.Resize(new SKImageInfo(100, 100), SKBitmapResizeMethod.Lanczos3))
            using (var image = SKImage.FromBitmap(resized)) image.Encode(SKEncodedImageFormat.Jpeg, 100).SaveTo(destinationFile);```

**Comment by mattleibow** [comment]
I had a look at this and the problem is because of a combination of factors. The specific combination is the webp codec and the http stream. For some reason, the webp codec tries to read the length of the stream (the jpeg does not need to) and the http stream does not have a length.

I will investigate a solution, but in the meantime you could create a buffer stream:

```csharp
using (var httpClient = new HttpClient())
using (var stream = await httpClient.GetStreamAsync(new Uri("http://www.gstatic.com/webp/gallery/2.webp")))
using (var buffer = new MemoryStream())
{
	await stream.CopyToAsync(buffer);
	using (var skBitmap = SKBitmap.Decode(buffer))
	using (var destinationFile = new FileStream("D:/desktop/teste_webp_format.jpg", FileMode.OpenOrCreate))
	using (var resized = skBitmap.Resize(new SKImageInfo(100, 100), SKBitmapResizeMethod.Lanczos3))
	using (var image = SKImage.FromBitmap(resized)) image.Encode(SKEncodedImageFormat.Jpeg, 100).SaveTo(destinationFile);
	{
	}
}
```

**Comment by smartbracker** [comment]
Thanks for reply!  
I'll use this for now.

**Comment by mattleibow** [comment]
This has been fixed in aab51650ac8311682f92226c4021bfafdcd9c394 and a58f62bfc99c8a1b74c43123508cd150f663e682
